image_tools/CMakeLists.txt

```diff
@@ -13,37 +13,63 @@ endif()
 
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(sensor_msgs REQUIRED)
 find_package(std_msgs REQUIRED)
-
+find_package(rcutils REQUIRED)
 find_package(OpenCV REQUIRED)
 
 include_directories(include)
 
-add_executable(cam2image
-  src/burger.cpp
-  src/cam2image.cpp
-  src/options.cpp
-)
-ament_target_dependencies(cam2image
+add_library(showimage_component SHARED
+  src/showimage.cpp)
+ament_target_dependencies(showimage_component
   "rclcpp"
   "sensor_msgs"
-  "std_msgs"
+  "std_msgs"    
+  "rclcpp_components"
   "OpenCV")
+rclcpp_components_register_nodes(showimage_component "image_tools::ShowImage")
 
-add_executable(showimage
-  src/options.cpp
-  src/showimage.cpp
-)
-ament_target_dependencies(showimage
+add_library(cam2image_component SHARED 
+  src/cam2image.cpp)
+ament_target_dependencies(cam2image_component
   "rclcpp"
   "sensor_msgs"
   "std_msgs"
-  "OpenCV")
+  "OpenCV"
+  "rclcpp_components")
+rclcpp_components_register_nodes(cam2image_component "image_tools::Cam2Image")
+
+add_executable(cam2image
```

Comment:
That should happen as part of the components CMake API - maybe optionally controlled by a flag - but the user code shouldn't need to do it explicitly.

Reply:
Correct me if I'm wrong, but this looks like something that is not implemented yet in the components CMake API. I believe this is the relevant ticket: https://github.com/ros2/rclcpp/issues/114

@skucheria I would add a TODO here (and above the other `add_executable`) stating that the components API should handle this for us.

Reply:
I think the referenced feature needs to land before this can be merged.